src/link/MachO/load_commands.zig

```diff
@@ -467,8 +467,30 @@ pub inline fn appleVersionToSemanticVersion(version: u32) std.SemanticVersion {
     };
 }
 
-pub fn inferSdkVersionFromSdkPath(path: []const u8) ?std.SemanticVersion {
-    const stem = std.fs.path.stem(path);
+pub fn inferSdkVersion(gpa: Allocator, comp: *const Compilation) ?std.SemanticVersion {
+    var arena_allocator = std.heap.ArenaAllocator.init(gpa);
+    defer arena_allocator.deinit();
+    const arena = arena_allocator.allocator();
+
+    const options = comp.bin_file.options;
+
+    const sdk_dir = switch (options.darwinSdkLayout) {
+        .none => unreachable,
+        .sdk => options.sysroot.?,
+        .vendored => std.fs.path.join(arena, &.{ comp.zig_lib_directory.path.?, "libc", "darwin" }) catch return null,
+    };
+
+    if (readSdkVersionFromSettings(arena, sdk_dir)) |ver| {
+        return parseSdkVersion(ver);
+    } else |_| {
+        if (options.darwinSdkLayout == .vendored) {
+            // vendored layout does not have versioned pathname
+            return null;
+        }
```

Comment:
I think we should panic here as technically this should be unreachable unless we forgot to include `SDKSettings.json` file, right?

Reply:
I think this should stand.

This is _after_ SDKSettings.json attempt. If that fails, we continue to older infer-from-pathname code. But it's a complete waste to do it if vendored, because it will always be null. This is just a optimization to avoid that needless check that will fail with null.

Reply:
Right, but my point is that we control the `vendored` path completely so I don't see how parsing the settings json can fail, and if it does it means we terminally failed and should fix it rather than fallback to parsing the path.

Reply:
And going further with this, I would like to know if we messed up in setting up vendored sdk rather than silently fail.

Reply:
ok I grok that standpoint now